src/Servers/Kestrel/Core/src/Internal/Http3/Http3Stream.cs

```diff
@@ -98,15 +98,39 @@ public Http3Stream(Http3Connection http3Connection, Http3StreamContext context)
 
         public void Abort(ConnectionAbortedException ex)
         {
-            Abort(ex, Http3ErrorCode.InternalError);
+            AbortCore(ex, Http3ErrorCode.InternalError);
         }
 
-        public void Abort(ConnectionAbortedException ex, Http3ErrorCode errorCode)
+        public void ResetAndAbort(ConnectionAbortedException ex, Http3ErrorCode errorCode)
+        {
+            // TODO - Keep track of whether stream has been aborted before?
+
+            Log.Http3StreamResetAbort(TraceIdentifier, errorCode, ex);
```

Comment:
Seems like the only difference here is that we are logging that we are sending a reset. Why can't these be combined with the normal Abort path with ResetAndAbort?